lemminx-maven/src/test/java/org/eclipse/lemminx/extensions/maven/participants/hover/MavenPropertyHoverTest.java

```diff
@@ -0,0 +1,70 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Red Hat Inc. and others.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.lemminx.extensions.maven.participants.hover;
+
+import static org.eclipse.lemminx.extensions.maven.utils.MavenLemminxTestsUtils.createDOMDocument;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.concurrent.ExecutionException;
+
+import org.eclipse.lemminx.dom.DOMDocument;
+import org.eclipse.lemminx.extensions.maven.NoMavenCentralExtension;
+import org.eclipse.lemminx.services.XMLLanguageService;
+import org.eclipse.lemminx.settings.SharedSettings;
+import org.eclipse.lsp4j.Hover;
+import org.eclipse.lsp4j.Position;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(NoMavenCentralExtension.class)
+public class MavenPropertyHoverTest {
+	private XMLLanguageService languageService;
+
+	@BeforeEach
+	public void setUp() throws IOException {
+		languageService = new XMLLanguageService();
+	}
+
+	@AfterEach
+	public void tearDown() throws InterruptedException, ExecutionException {
+		languageService.dispose();
+		languageService = null;
+	}
+	
+	@Test
+	public void testHpverForVariablePropertyDefinedInSameDocument()
+			throws IOException, InterruptedException, ExecutionException, URISyntaxException {
+		DOMDocument document = createDOMDocument("/pom-with-properties-in-parent-for-definition.xml", languageService);
+		Hover hover = languageService.doHover(document, new Position(28, 15), new SharedSettings());
+		assertNotNull(hover);
+		String hoverContents = hover.getContents().getRight().getValue();
+		assertNotNull(hoverContents);
+		assertTrue(hoverContents.contains("anotherProperty"));
+		assertTrue(hoverContents.contains("$"));
+		assertTrue(hoverContents.contains("org.test:child:0.0.1-SNAPSHOT"));
```

Comment:
Can we also check the link is present?

Reply:
Yes. Using `XMLAssert.assertHover(...)` solves it.